Shared/Samples/Download preplanned map area/DownloadPreplannedMapAreaView.Model.swift

```diff
@@ -165,45 +165,49 @@ extension DownloadPreplannedMapAreaView.Model {
     }
 }
 
-/// An object that encapsulates state about an offline map.
-class OfflineMapModel: ObservableObject, Identifiable {
-    /// The preplanned map area.
-    let preplannedMapArea: PreplannedMapArea
-    
-    /// The task to use to take the area offline.
-    let offlineMapTask: OfflineMapTask
-    
-    /// The directory where the mmpk will be stored.
-    let mmpkDirectory: URL
-    
-    /// The currently running download job.
-    @Published private(set) var job: DownloadPreplannedOfflineMapJob?
-    
-    /// The result of the download job.
-    @Published private(set) var result: Result<MobileMapPackage, Error>?
-    
-    init?(preplannedMapArea: PreplannedMapArea, offlineMapTask: OfflineMapTask, temporaryDirectory: URL) {
-        self.preplannedMapArea = preplannedMapArea
-        self.offlineMapTask = offlineMapTask
+extension DownloadPreplannedMapAreaView {
+    /// An object that encapsulates state about an offline map.
+    class OfflineMapModel: ObservableObject, Identifiable {
+        /// The preplanned map area.
+        let preplannedMapArea: PreplannedMapArea
         
-        if let itemID = preplannedMapArea.portalItem.id {
-            self.mmpkDirectory = temporaryDirectory
-                .appendingPathComponent(itemID.rawValue)
-                .appendingPathExtension("mmpk")
-        } else {
-            return nil
+        /// The task to use to take the area offline.
+        let offlineMapTask: OfflineMapTask
+        
+        /// The directory where the mmpk will be stored.
+        let mmpkDirectory: URL
+        
+        /// The currently running download job.
+        @Published private(set) var job: DownloadPreplannedOfflineMapJob?
+        
+        /// The result of the download job.
+        @Published private(set) var result: Result<MobileMapPackage, Error>?
+        
+        init?(preplannedMapArea: PreplannedMapArea, offlineMapTask: OfflineMapTask, temporaryDirectory: URL) {
+            self.preplannedMapArea = preplannedMapArea
+            self.offlineMapTask = offlineMapTask
+            
+            if let itemID = preplannedMapArea.portalItem.id {
+                self.mmpkDirectory = temporaryDirectory
+                    .appendingPathComponent(itemID.rawValue)
+                    .appendingPathExtension("mmpk")
+            } else {
+                return nil
+            }
         }
-    }
-    
-    deinit {
-        Task { [job] in
-            // Cancel any outstanding job.
-            await job?.cancel()
+        
+        deinit {
+            Task { [job] in
+                // Cancel any outstanding job.
+                await job?.cancel()
+            }
         }
     }
 }
 
-extension OfflineMapModel: Hashable {
+extension DownloadPreplannedMapAreaView.OfflineMapModel: Hashable {
```

Comment:
Not sure if there's anything that can be done about it at this point, but the implementation of the requirements of these protocols do not fit the requirements as outlined in the documentation. Most likely `ObjectIdentifier` (which adopts `Equatable` and `Hashable`) should have been used (as evidenced by the implementation of `hash(into:)`) instead of the model itself.

Also, FYI: equatable and hashable are really two separate protocols and are typically adopted using two separate extensions. When adopting a protocol via an extension, it is helpful to only include the requirements for that protocol in that extension, and nothing else. The typealias is unnecessary. If `Self` can't be used, then just use the full type name, even if it is long. After all, that is how the compiler synthesizes it.

Reply:
I agree the original implementation isn't the best, but I will leave it as is for now to avoid refactoring the sample in this PR. I did remove the typealias, though, and broke the extension into two, one per conformance. Let me know if you had something different in mind.

Reply:
Sounds good. Thanks!